st3/sublime_lib/show_selection_panel.py

```diff
@@ -0,0 +1,68 @@
+NOT_GIVEN = {}
+
+
+def show_selection_panel(
+    window,
+    items,
+    *,
+    flags=None,
+    labels=str,
+    selected=NOT_GIVEN,
+    on_select=None,
+    on_cancel=None,
+    on_highlight=None
+):
+    """Open a quick panel in the given window to select an item from a list.
+
+    :argument window: The :class:`sublime.Window` in which to show the panel.
+
+    :argument items: A :class:`list` of items to choose from.
+
+    Optional keyword arguments:
+
+    :argument flags: A bitwise OR of :const:`sublime.MONOSPACE_FONT` and
+    :const:`sublime.KEEP_OPEN_ON_FOCUS_LOST`.
+
+    :argument labels: A function taking elements of `items` to string labels.
```

Comment:
`items` (the parameter to `show_quick_panel`) may be a list of strings for multi-line display. Although we don't *have* to support it, it'd be a waste imo. So, a "label" can also be a list of strings, but their length should be consistent due to ST bugs.

Furthermore, I suggest an alternative use case where you already have the labels to show in the panel and the values to resolve to in a collection (or two). A list of two-tuples (or just two-item collections) sounds like a good compromise, where you'd have a list like follows:
```py
items = [
    ("label1", "value1"),
    ("label2", "value2"),
]
```
Allowing labels to be a list would be an easy way for `labels, values = zip(*items)` and then using these two lists in the `show_selection_panel` call.
Alternatively, passing a value->label map here might also seem useful, but that'd need some exploration. All values must be hashable in that case.

Also not so fond of the parameter name, if only functions are accepted. `labels` seems to imply it would expect a collection of labels, not a function that maps items to labels.

TL;DR
- Allow a list to be passed as the labels.
- Still allow a function to be passed. Useful for the default value of `str`, for example.
- Explore allowing a dict to be passed that has both values and labels; still need a separate param for order? value->label or label->value? different param?